fdbclient/NativeAPI.actor.cpp

```diff
@@ -1759,14 +1796,19 @@ ACTOR Future<Standalone<RangeResultRef>> getRangeFallback( Database cx, Version
 	return r;
 }
 
-void getRangeFinished(Reference<TransactionLogInfo> trLogInfo, double startTime, KeySelector begin, KeySelector end, bool snapshot,
+void getRangeFinished(Database cx, Reference<TransactionLogInfo> trLogInfo, double startTime, KeySelector begin, KeySelector end, bool snapshot,
 	Promise<std::pair<Key, Key>> conflictRange, bool reverse, Standalone<RangeResultRef> result)
 {
+	int64_t bytes = 0;
+	for(const KeyValueRef &kv : result) {
+		bytes += kv.key.size() + kv.value.size();
+	}
```

Comment:
Based on profilings Mako did on a similar change while implementing the read hot detection, this loop will dramatically slow down the system if it's on the critical path, i.e. on every getRange request code path.

Reply:
I ended up doing this: https://github.com/apple/foundationdb/blob/master/fdbserver/storageserver.actor.cpp#L1434

Reply:
It seems like the linked code does the same thing as I'm doing just a few lines above the linked line:

```
int64_t totalByteSize = 0;
for (int i = 0; i < r.data.size(); i++) {
	totalByteSize += r.data[i].expectedSize();
}
```

Are you saying that the current code in storage server is expensive, or this is what you settled on that works well?

The loop that I'm running is also something that I think we already do on the clients to handle limits, so that should serve to bound how expensive it is, relatively speaking. If it turns out that doing this is expensive, though, then we should probably be thinking about trying to update the logic not to need to iterate through the list to get its byte size.

Reply:
I lost track of my own code LOL. My bad.

The expensive part actually is not the loop counting the size, it's the loop, that I removed due to the perf regression, that inserting them into the sampler. Your code here would be fine.